### PR TITLE
Initial subnet contract implementation

### PR DESCRIPTION
**Pull request by fyrchik**
Support put/get/delete methods.

Related #122 .
Depends on #168 .

### CODE REVIEW

File: subnet/config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+safemethods: ["version"]
+permissions:
+  - methods: ["update"]
+events:
+  - name: SubnetPut
+    parameters:
+      - name: ownerKey
+        type: PublicKey
+      - name: info
+        type: ByteArray
```
Comment (alexvanin) [inline subnet/config.yml:11]: Go for new line

File: subnet/subnet_contract.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		id := common.InvokeID([]interface{}{ownerKey, info}, []byte("put"))
+		n := common.Vote(ctx, id, nodeKey)
+		if n < threshold {
+			return
+		}
+
+		common.RemoveVotes(ctx, id)
+	} else {
+		if !runtime.CheckWitness(ownerKey) {
+			panic("put: witness check failed")
```
Comment (alexvanin) [inline subnet/subnet_contract.go:73]:
```suggestion
			panic("put: owner witness check failed")
```

File: subnet/subnet_contract.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+		multiaddr := common.AlphabetAddress()
+		if !runtime.CheckWitness(multiaddr) {
+			panic("put: alphabet witness check failed")
+		}
+	}
+
+	stKey := append([]byte{ownerPrefix}, id...)
+	if storage.Get(ctx, stKey) != nil {
+		panic("put: " + ErrAlreadyExists)
+	}
```
Comment (alexvanin) [inline subnet/subnet_contract.go:85]: Put this check before `if notaryDisable` condition. Early checks before `runtime.Notify` are appreciated by the clients and it saves a bit of wasted GAS in notary disabled environment.

File: subnet/subnet_contract.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+	ctx := storage.GetContext()
+	notaryDisabled := storage.Get(ctx, notaryDisabledKey).(bool)
+	if notaryDisabled {
+		alphabet := common.AlphabetNodes()
+		nodeKey := common.InnerRingInvoker(alphabet)
+		if len(nodeKey) == 0 {
+			if !runtime.CheckWitness(ownerKey) {
+				panic("put: witness check failed")
```
Comment (carpawell) [inline subnet/subnet_contract.go:57]:
```suggestion
				panic("put: owner witness check failed")
```

File: subnet/subnet_contract.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	storage.Put(ctx, stKey, info)
+}
+
+// Get returns info about subnet with the specified id.
+func Get(id []byte) []byte {
+	ctx := storage.GetContext()
+	key := append([]byte{infoPrefix}, id...)
+	raw := storage.Get(ctx, key)
+	if raw == nil {
+		panic(ErrNotExist)
```
Comment (carpawell) [inline subnet/subnet_contract.go:98]: should it be panic when clients check if the network hasn't been deleted?

Comment (fyrchik) [reply subnet/subnet_contract.go:98]: We can try to return `Null` but the we should also require non-empty info in `Put`, I think.

**Review by alexvanin — APPROVED** [review]

**Review by carpawell — APPROVED** [review]

File: subnet/subnet_contract.go (hunk 5 above)
Comment (carpawell) [reply subnet/subnet_contract.go]: just thought that `Null` or smth like that can be more user-friendly than checking the message of the contract panic (I just think that getting subnetwork can be the usual call, like an healthcheck of the network, to prove that the owner has not deleted the network)

but if panic is a common practice of returning errors from contracts, it's ok

**Pull request merged**